R/validate_format.R

```diff
@@ -0,0 +1,227 @@
+#'  column `value`.
+#' @param intervals a vector of `numeric` values indicating which central
+#'  prediction interval levels to plot. `NULL` means no interval levels.
+#' @param list_intervals named list of accepted intervals accepted
+#' @param max_model_model_id numeric, if model_output_data contains more than
+#'  this number of unique `"model_id"`, the intervals will be reduced to
+#'  only one value (the maximal)
+#'
+#' @noRd
+interval_validation <- function(model_output_data, intervals, list_intervals,
+                                max_model_id = 5) {
+  if (any(!intervals %in% names(list_intervals))) {
+    cli::cli_warn(c("!" = "{.arg intervals} should correspond to one or
+                      multiple of these possible values
+                      {.val {names(list_intervals)}}.
+                      Only the matching value(s) will be used (if no matching
+                      value, the default will be used)."))
+    intervals <- intervals[intervals %in% names(list_intervals)]
+    if (length(intervals) == 0) {
+      intervals <- as.character(c(.5, .8, .95))
+    }
+  }
+  if (length(unique(model_output_data[["model_id"]])) > max_model_id &&
+        length(intervals) > 1) {
+    intervals <- max(intervals)[1]
+    cli::cli_warn(c("!" = "{.arg model_output_data} contains more than
+                    {.val {max_model_id}} models, the plot will be reduced to
+                    show only one interval (the maximum interval value):
+                    {.val {intervals}}"))
+  }
+  return(intervals)
+}
+
+#' Validate `"ensembles"` parameters format
+#'
+#' Validate `ens_color`, `ens_name` format: both should be set to `NULL` or to
+#' a non `NULL` value.
+#'
+#'@param ens_color a `character` string of a color name, if not NULL, will be
+#' use as color for the model name associated with the parameter `ens_name`
+#' (both parameter need to be provided)
+#'@param ens_name a `character` string of a model name, if not NULL, will be
+#' use to change the color for the model name, associated with the parameter
+#' `ens_color`(both parameter need to be provided)
+#'
+#' @noRd
+ensemble_validation <- function(ens_color, ens_name) {
+  if (is.null(ens_color) + is.null(ens_name) == 1) {
+    cli::cli_abort(c("x" = "Both {.arg ens_color} and {.arg ens_name} should
+                     be set to a non NULL value"))
+  }
+}
+
+#' Validate `"output_type_id"` values
+#'
+#' Validate model output data contains the expected `"output_type_id"` values
+#'
+#' @param model_output_data a `model_out_tbl` object, containing all the
+#'  required columns including a column containing date information and a
+#'  column `value`.
+#' @param exp_value expected value required in `model_output_data`
+#' `"output_type_id"` column
+#'
+#' @noRd
+output_type_validation <- function(model_output_data, exp_value) {
+  if (!class(exp_value) %in% class(model_output_data$output_type_id)) {
```

Comment:
Thanks for the comment, I updated the documentation. 

So currently, as the function is only accepting quantile et median output type, it's expected to have numeric in the `output_type_id` column. `exp_value` is the numeric vector with the the `output_type_id` value that are going to be used for the plot: for example: `0.5`, '0.975`, 0.025`, etc. The `exp_value` is created internally in the `plot_step_ahead_model_output()` function.  I force the column to be in the `numeric` because it causes issue if it's not. 

Does that answer your question?

Reply:
Sort of. I guess my question is why compare classes of the two objects and not just check whether output type Id is numeric and if not coerce it?

Reply:
To be honest, I was thinking of when we will have other format (for example with sample) but as we are not there yet. I will simplify it. Thank you again for noticing it!